pkg/operator/ceph/csi/csi.go

```diff
@@ -176,9 +176,13 @@ func (r *ReconcileCSI) setParams(ver *version.Info) error {
 	}
 
 	CSIParam.EnableCSIAddonsSideCar = false
-	if strings.EqualFold(k8sutil.GetValue(r.opConfig.Parameters, "CSI_ENABLE_CSIADDONS", "false"), "true") {
+	_, err = r.context.ApiExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(r.opManagerContext, "csiaddonsnode.csiaddons.openshift.io", metav1.GetOptions{})
+	if err == nil {
 		CSIParam.EnableCSIAddonsSideCar = true
 	}
+	if strings.EqualFold(k8sutil.GetValue(r.opConfig.Parameters, "CSI_ENABLE_CSIADDONS", ""), "false") {
+		CSIParam.EnableCSIAddonsSideCar = false
+	}
```

Comment:
I think we are missing one case here

* Enable the sidecar first and install the csiaddons CRD after sometime.

With current change we will never enable the sidecar

Reply:
> Enable the sidecar first and install the csiaddons CRD after sometime.

any suggestion on how to check the time after which we should install csiaddons?

Reply:
done